charmcraft/models/config.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Charmcraft Juju Config pydantic model."""
+
+import pydantic
+
+from typing import Optional, Dict
+
+from charmcraft.models.basic import ModelConfigDefaults
+
+
+class JujuConfig(ModelConfigDefaults):
+    """Juju configs for charms.
+
+    See also: https://juju.is/docs/sdk/config
+    """
+
+    options: Optional[Dict[str, Dict]]
+    legacy: bool = False
+
+    @pydantic.validator("options", pre=True)
+    def validate_actions(cls, options, values):
+        """Verify options section."""
+        for option in options:
+            if not isinstance(options[option], dict):
+                raise ValueError(f"'{option}' is not a dictionary")
+
+            if not set(options[option].keys()).issubset({"description", "type", "default"}):
+                raise ValueError(f"'{option}' has an invalid key")
+
+            if "type" not in options[option]:
+                raise ValueError(f"'{option}' is missing a type")
+
+            if options[option]["type"] not in ["string", "int", "float", "boolean"]:
+                raise ValueError(f"'{option}' has an invalid type")
```

Comment:
```suggestion
                invalid_type = options[option]["type"]
                raise ValueError(f"'{option}' type {invalid_type!r} is invalid. Should be one of ['string', 'int', 'float', 'boolean']")
```

or something like that :-)